### PR TITLE
mpdel: add evil keybind

### PR DESCRIPTION
**Pull request by Vonfry**
Several questions:

- [x] how to write license in the header
- [ ] some keybinds may not suitable
- [X] github ci seems cannot find some function, how to fix it ?

Fix: #238

### CODE REVIEW

**Review by condy0919 — COMMENTED** [review]
Some nitpicks

File: Cask [hunk 1]
```diff
@@ -9,4 +9,5 @@
  (depends-on "ert-runner")
  (depends-on "package-lint")
  (depends-on "annalist")
- (depends-on "magit"))
+ (depends-on "magit")
+ (depends-on "mpdel"))
```
Comment (jojojames) [inline Cask:13]: Probably don't need this unless you're adding tests.

File: modes/mpdel/evil-collection-mpdel.el [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+;; Evil bindings for mpdel.
+
+;;; Code:
+
+(require 'mpdel nil t)
+(require 'evil-collection)
+
+(defvar mpdel-core-map)
+
+(defconst evil-collection-mpdel-maps '(mpdel-browser-mode mpdel-playlist-mode
```
Comment (jojojames) [inline modes/mpdel/evil-collection-mpdel.el:36]: These should be mode maps.

File: modes/mpdel/evil-collection-mpdel.el [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                                       mpdel-song-mode mpdel-browser-mode
+                                       mpdel-playlist-current-playlist-mode
+                                       mpdel-playlist-stored-playlist-mode))
+
+;;;###autoload
+(defun evil-collection-mpdel-setup ()
+  "Set up `evil' bindings for `mpdel'."
+
+  (evil-collection-set-readonly-bindings 'mpdel-core-map)
+  (dolist (mode evil-collection-mpdel-maps)
```
Comment (jojojames) [inline modes/mpdel/evil-collection-mpdel.el:46]: Can probably define a different variable to do this loop.

File: modes/mpdel/evil-collection-mpdel.el [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    "R"  #'mpdel-core-replace-stored-playlist
+    "i"  #'mpdel-core-insert-current-playlist
+    "gD" #'mpdel-core-dired
+    "gA" #'mpdel-core-open-artists
+    "gL" #'mpdel-core-open-stored-playlists
+    "ss" #'mpdel-core-search-by-title
+    "sl" #'mpdel-core-search-by-album
+    "sr" #'mpdel-core-search-by-artist
+    "gp" #'navigel-open-parent
+    "j"  #'next-line
```
Comment (jojojames) [inline modes/mpdel/evil-collection-mpdel.el:64]: evil-next-line etc

File: modes/mpdel/evil-collection-mpdel.el [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    "k"  #'previous-line
+    "+"  #'mpdel-core-volume-increase
+    "-"  #'mpdel-core-volume-decrease
+    "C"  #'libmpdel-connect-profile
+
+    "gb" #'mpdel-browser-open
+
+    "go" #'mpdel-song-open
+    "gr" #'mpdel-song-refresh
+    "I"  #'mpdel-song-play
```
Comment (jojojames) [inline modes/mpdel/evil-collection-mpdel.el:74]: This doesn't seem super intuitive but I've never used this package so maybe it's OK.

Consider maybe adding a wrapper function for toggle play? Maybe that'd be more appropriate for the original package.

File: modes/mpdel/evil-collection-mpdel.el (hunk 2 above)
Comment (Vonfry) [reply modes/mpdel/evil-collection-mpdel.el:36]: Sorry for using wrong name. These modes are passed to `evil-set-initial-state`. I will rename it to `evil-collection-mpdel-modes`

File: Cask (hunk 1 above)
Comment (Vonfry) [reply Cask:13]: No new test is added, but the ci is failed at first time opening the pr without these lines.

File: modes/mpdel/evil-collection-mpdel.el (hunk 2 above)
Comment (jojojames) [reply modes/mpdel/evil-collection-mpdel.el:36]: You'll want another const holding all the relevant mode maps.

Comment (Vonfry) [reply modes/mpdel/evil-collection-mpdel.el:36]: The `mpdel-core-map` is the original one used by mpdel. It doesn't contain `mode` in its name.

File: Cask (hunk 1 above)
Comment (Vonfry) [reply Cask]: I pushed a new commit without this, and the ci failed.

File: modes/mpdel/evil-collection-mpdel.el (hunk 5 above)
Comment (Vonfry) [reply modes/mpdel/evil-collection-mpdel.el:76]: > Consider maybe adding a wrapper function for toggle play?

The key `p` is used for that. The three functions are not used to toggle play. `song-open` is used to open a buffer to show current playing song information. `song-refresh` is used to refresh buffer content. `song-play` is used to play a single song under cursor in browser.

**Comment by Vonfry** [comment]
ping @jojojames

**Comment by condy0919** [comment]
I'm not mpdel user, but maybe we can merge and let it evolve? thoughts? @jojojames

**Comment by jojojames** [comment]
@condy0919 @Vonfry Forgot about this one, thanks for the reminder. Looks OK to merge.

**Pull request merged**